### ISSUE TITLE
Link always jumps to the top of the file when a header is nested in it

### ISSUE THREAD

**Issue by Mysara1993** [wkhtmltopdf/wkhtmltopdf#3821]
HTML allows the nesting of a header inside a link and the whole header block works as the link.
But in that case after transforming to PFD using wkhtmltopdf, the link always jumps to the top of the PDF file.
Below is an html snippet with both cases (link nested in header and header nested in link)
`<h1>Example page</h1>

<h3><a name="a001">a001</a></h3>
<p>paragraph text ...</p>

<h3><a name="a002" id="a002">a002</a></h3>
<p>paragraph text ...</p>

<a name="a003" id="a003"><h3>a003</h3></a>
<p>paragraph text ...</p>

<p><a href="#a002">Jump to a002</a></p>
<p><a href="#a003">Jump to a003</a></p>`

Thank you!

**Comment by benedekh** [comment]
Have you checked [this solution](https://github.com/wkhtmltopdf/wkhtmltopdf/issues/3486)?

**Comment by Mysara1993** [comment]
Yes my anchor is not empty in the code. I am saying that it works fine when the link is in the header but doesn't when the header is inside the link.
![image](https://user-images.githubusercontent.com/29484585/36473697-a18cb1fc-16f5-11e8-90ea-16ed4119e621.png)
